src/torrent_info.cpp

```diff
@@ -967,8 +966,6 @@ namespace libtorrent
 
 	void torrent_info::unload()
 	{
-		TORRENT_ASSERT(m_info_section.unique());
```

Comment:
`(std|boost)::shared_ptr` it's not specialized for arrays, it works but you need to provide a custom deleter (easy) and there is no operator `[]`